framework/source/class/qx/theme/manager/Decoration.js

```diff
@@ -122,8 +122,11 @@ qx.Class.define("qx.theme.manager.Decoration",
       // create and add a CSS rule
       var css = "";
       var styles = instance.getStyles(true);
-      for (var key in styles) {
-
+      
+      // Sort the styles so that more specific styles come after the group styles, 
+      //  eg background-color comes after background
+      var t = this;
+      Object.keys(styles).sort().forEach(function(key) {
```

Comment:
According to https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Map the keys in the map get returned in insertion order ... But since I can not see how a lexical sorting should have a negative influence, I would urge to have the keys sorted as you suggest and add appropriate documentation ... that would certainly improve the current rather random situation.